ui/src/dashboards/reducers/dashboards.ts

```diff
@@ -22,83 +42,66 @@ export const dashboardsReducer = (
 ): DashboardsState => {
   return produce(state, draftState => {
     switch (action.type) {
-      case ActionTypes.SetDashboards: {
-        const {list, status} = action.payload
-
-        draftState.status = status
-        if (list) {
-          draftState.list = list
-        }
+      case SET_DASHBOARDS: {
+        setResource<Dashboard>(draftState, action, ResourceType.Dashboards)
 
         return
       }
 
-      case ActionTypes.RemoveDashboard: {
-        const {id} = action.payload
-        draftState.list = draftState.list.filter(l => l.id !== id)
+      case REMOVE_DASHBOARD: {
+        removeResource<Dashboard>(draftState, action)
 
         return
       }
 
-      case ActionTypes.SetDashboard: {
-        const {dashboard} = action.payload
-        draftState.list = _.unionBy([dashboard], state.list, 'id')
+      case SET_DASHBOARD: {
+        const {schema} = action
+        const {entities, result} = schema
+
+        draftState.byID[result] = entities.dashboards[result]
+        const exists = draftState.allIDs.find(id => id === result)
+
+        if (!exists) {
```

Comment:
is there a chance, however small, that the id it finds could be `0`, and this could be falsy? or will it always be a stringy uuid?

Reply:
nope.  they are all strings of the uuid v4